src/QsCompiler/CompilerOptimizations/CallableInlining.fs

```diff
@@ -81,12 +81,30 @@ type private InliningInfo = {
     body: QsScope
 }
 
+/// A scope transformation that substitutes type parameters according to the given dictionary
+type private ReplaceTypeParams(typeParams: ImmutableDictionary<QsTypeParameter, ResolvedType>) =
+    inherit ScopeTransformation()
+
+    override __.Expression = { new ExpressionTransformation() with
+        override __.Type = { new ExpressionTypeTransformation() with
+            override this.onTypeParameter tp =
+                if typeParams.ContainsKey tp then
```

Comment:
You probably want to explicitly ignore range information here and only compare origin and name.

Reply:
Fixed in latest commit